src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -387,10 +366,10 @@ RandomFourierGaussPreproc::apply_to_matrix(SGMatrix<float64_t> matrix)
 			SGVector<float64_t> a(
 			    matrix.matrix + vec * num_features, cur_dim_input_space, false);
 			SGVector<float64_t> b(
-			    randomcoeff_multiplicative + od * cur_dim_input_space,
+			    random_weights + od * cur_dim_input_space,
 			    cur_dim_input_space, false);
 			res(od, vec) =
-			    val * cos(randomcoeff_additive[od] + linalg::dot(a, b));
+			    val * cos(random_offset[od] + linalg::dot(a, b));
```

Comment:
you can easily do a matrix-vector product here, rather than looping over the vectors